SyncTests/CryptoTests.swift

```diff
@@ -80,4 +81,63 @@ class CryptoTests: XCTestCase {
             XCTFail("Encrypt failed.")
         }
     }
+    
+    func testSignVerify() {
+        let cleartext = Bytes.decodeBase64(cleartextB64)
+        
+        // DSA
+        let dsa = DSAKeyPair.generateKeyPairWithSize(1024)
+        XCTAssertNotEqual(dsa, nil)
+        XCTAssertNotEqual(dsa.privateKey, nil)
+        XCTAssertNotEqual(dsa.publicKey, nil)
+        
+        let sigDSA = dsa.privateKey.signMessage(cleartext)
+        XCTAssertNotEqual(sigDSA, nil)
+        
+        let verDSA = dsa.publicKey.verifySignature(sigDSA, againstMessage: cleartext)
+        XCTAssertTrue(verDSA)
+        
+        // RSA
+        let rsa = DSAKeyPair.generateKeyPairWithSize(1024)
+        XCTAssertNotEqual(rsa, nil)
+        XCTAssertNotEqual(rsa.privateKey, nil)
+        XCTAssertNotEqual(rsa.publicKey, nil)
+        
+        let sigRSA = rsa.privateKey.signMessage(cleartext)
+        XCTAssertNotEqual(sigRSA, nil)
+
+        let verRSA = rsa.publicKey.verifySignature(sigRSA, againstMessage: cleartext)
+        XCTAssertTrue(verRSA)
+        
+        // ECDSA
+        let ecdsa = ECDSAKeyPair.generateKeyPairForGroup(ECDSAGroup.P256)
+        XCTAssertNotEqual(ecdsa, nil)
+        XCTAssertNotEqual(ecdsa.privateKey, nil)
+        XCTAssertNotEqual(ecdsa.publicKey, nil)
+        
+        let sigECDSA = ecdsa.privateKey.signMessage(cleartext)
+        XCTAssertNotEqual(sigECDSA, nil)
+        
+        let verECDSA = ecdsa.publicKey.verifySignature(sigECDSA, againstMessage: cleartext)
+        XCTAssertTrue(verECDSA)
+
+        let privBytes = ecdsa.privateKey.BinaryRepresentation()
+        XCTAssertNotNil(privBytes)
+        
+        let privECDSA2 = ECDSAPrivateKey(binaryRepresentation: privBytes, group: .P256)
+        XCTAssertNotNil(privECDSA2)
+        let sigECDSA2 = privECDSA2.signMessage(cleartext)
+        let verECDSA2 = ecdsa.publicKey.verifySignature(sigECDSA2, againstMessage: cleartext)
+        XCTAssertTrue(verECDSA2)
+
+        let pubBytes = ecdsa.publicKey.BinaryRepresentation()
+        let pubECDSA3 = ECDSAPublicKey(binaryRepresentation: pubBytes, group: .P256)
+        XCTAssertNotNil(pubECDSA3)
+        let verECDSA3 = pubECDSA3.verifySignature(sigECDSA, againstMessage: cleartext)
```

Comment:
Nit: It's sort of weird that this `verifySignature` doesn't check `sigECDSA2`, but it's NBD.